src/objects/file.py

```diff
@@ -11,93 +12,194 @@
+
+        :return: uid of this file.
+        '''
         if self._uid is None and self.binary is not None:
             self._uid = create_uid(self.binary)
         return self._uid
 
     @uid.setter
-    def uid(self, new_uid):
+    def uid(self, new_uid: str):
         if self._uid is not None:
             logging.warning('uid overwrite: Uid might not be related to binary data anymore: {} -> {}'.format(self._uid, new_uid))
         self._uid = new_uid
 
-    def get_hid(self, root_uid=None):
+    def get_hid(self, root_uid: str = None) -> str:
         '''
-        return a human readable identifier
+        Get a human readable identifier for the given file.
+        This usually is the file name for extracted files.
+        As files can have different names across occurences, uid of a specific root object can be specified.
+
+        :param root_uid: (Optional) root uid to base HID on.
+        :return: String presenting human readable identifier for file.
         '''
         if root_uid is None:
             root_uid = self.get_root_uid()
         virtual_path = self.get_virtual_paths_for_one_uid(root_uid=root_uid)[0]
         return get_top_of_virtual_path(virtual_path)
 
-    def create_from_file(self, file_path):
+    def _create_from_file(self, file_path: str):
         self.set_binary(get_binary_from_file(file_path))
-        self.set_file_path(file_path)
+        self.create_binary_from_path()
+
+    def add_included_file(self, file_object) -> None:
+        '''
+        This functions adds a file to this objects list of included files.
+        The function also takes care of a number of meta information for the child object:
 
-    def add_included_file(self, file_object):
+        * `parents`: Adds the uid of this file to the parents field of the child.
+        * `root_uid`: Sets the root uid of the child as this files uid.
+        * `depth`: The child inherits the unpacking depth from this file, incremented by one.
+        * `scheduled_analysis`: The child inherits this file's scheduled analysis.
+        * `virtual_file_path`: Sets a new virtual_file_path for the child, being <this_files_current_vfp|child_path>.
+
+        :param file_object: File that was extracted from the current file
+        '''
         file_object.parents.append(self.uid)
         file_object.root_uid = self.root_uid
         file_object.add_virtual_file_path_if_none_exists(self.get_virtual_paths_for_one_uid(root_uid=self.root_uid), self.uid)
         file_object.depth = self.depth + 1
         file_object.scheduled_analysis = self.scheduled_analysis
         self.files_included.add(file_object.uid)
 
-    def add_virtual_file_path_if_none_exists(self, parent_paths, parent_uid):
+    def add_virtual_file_path_if_none_exists(self, parent_paths: List[str], parent_uid: str) -> None:
+        '''
+        Add vfps to this file based on an existing list of paths on the parent and the parent's uid as root.
```

Comment:
I'm sure you were sick of writing "virtual file paths" at this point, but since this could be the first thing someone reads, we should probably not use an acronym here